src/tasks/LibraryBuilder/LibraryBuilder.cs

```diff
@@ -145,10 +168,14 @@ private void GatherAotSourcesObjects(StringBuilder aotSources, StringBuilder aot
 
                 if (symbolsAdded > 0)
                 {
-                    exportedAssemblies.Add(Path.GetFileName(compiledAssembly.Path));
+                    exportedAssemblies.Add(Path.GetFileNameWithoutExtension(compiledAssembly.Path));
                 }
             }
         }
+        if (exportedAssemblies.Count == 0)
+        {
+            throw new LogAsErrorException($"None of the compiled assemblies contain exported symbols. Resulting shared library would be unusable.");
```

Comment:
Is this only for the shared library scenario or the static one as well? Should we really limit this? You could build a library and handle all embedding on your own and still have a lot of value with library builder producing a self-contained library for you.

Reply:
Didn't realize that the static library version would be okay without any exported methods from `UnmanagedCallersOnly` decorated managed methods. I think we figured that if there were no native-to-managed wrappers from calling exported functions then no `mono_threads_attach_coop` would be called to try to initialize the runtime. For those building and working with static library, would they be able to initialize the runtime without going through that route? 

A general question, would it be obvious how to handle the embedding and have a call to initialize the runtime? Should we have some doc to walkthrough what would need to be done for both scenarios of shared or static built?

I'll add in `&& IsSharedLibrary` to the condition to make it shared library specific.

Reply:
So you have different scenarios:

1. Build a shared library, since that will only export unmanagedcallers entrypoint the library can't be used unless we have those exports and in that case we need to use the runtime_init_callback default or custom.
2. Build a static library, now all symbols exported from runtime are still public as well as symbols exported using unmanagedcallers entrypoint. User might still want to have the runtime_init_callback as default or custom since they don't want to initialize the runtime, just call the entrypoints exported using unmangedcallers entrypoint.
3. Build a static library, but since we have all runtime API's available, do full runtime init on your own, before calling any entrypoints or using mono_runtime_invoke to call any managed method, so no need to unmanagedcallers entry points and user don't want any auto init logic in this scenario, so needs to be turned off. Quite possible that this mode will be used by iOS/Android SDK to get total control of runtime init as well as how the library will be packaged and distributed.